packages/fuel-indexer/src/executor.rs

```diff
@@ -683,38 +770,29 @@ impl Executor for WasmIndexExecutor {
         let ptr = arg.get_ptr();
         let len = arg.get_len();
 
-        let res = timeout(
-            Duration::from_secs(self.timeout),
-            spawn_blocking({
-                let store = self.store.clone();
-                move || {
-                    let mut store_guard =
-                        tokio::runtime::Handle::current().block_on(store.lock());
-                    fun.call(&mut store_guard, ptr, len)
-                }
-            }),
-        )
-        .await;
-
-        match res {
-            Err(e) => {
-                error!("WasmIndexExecutor handle_events timed out: {e:?}.");
-                let _ = self.db.lock().await.revert_transaction().await?;
-                return Err(IndexerError::from(e));
+        let res = spawn_blocking({
+            let store = self.store.clone();
+            move || {
+                let mut store_guard =
+                    tokio::runtime::Handle::current().block_on(store.lock());
+                fun.call(&mut store_guard, ptr, len)
             }
-            Ok(Err(e)) => {
-                error!("WasmIndexExecutor handle_events failed: {e:?}.");
+        })
+        .await?;
+
+        if let Err(e) = res {
+            if e.clone().to_trap() == Some(TrapCode::UnreachableCodeReached)
+                && self.metering_points_exhausted().await
```

Comment:
Just to be clear: Reaching the metering limit results in `TrapCode::UnreachableCodeReached` and never anything else, yeah?